pyproject.toml

```
@@ -0,0 +1,67 @@
+[build-system]
+requires = ["setuptools~=65.3", "wheel~=0.37", "setuptools_scm[toml]~=7.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "ixbrl-viewer"
+dynamic = ['version']
+readme = "README.md"
+license = {text = "Apache-2.0"}
+description = "The Arelle iXBRL Viewer allows iXBRL reports to be viewed interactively in a web browser."
+authors = [
+    {name = "arelle.org", email = "support@arelle.org"}
+]
+keywords = ["xbrl"]
+classifiers = [
+    'Development Status :: 5 - Production/Stable',
+    'Intended Audience :: End Users/Desktop',
+    'Intended Audience :: Developers',
+    'License :: OSI Approved :: Apache Software License',
+    'Natural Language :: English',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.7',
+    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
+    'Operating System :: OS Independent',
+    'Topic :: Text Processing :: Markup :: XML'
+]
+requires-python = ">=3.7"
+dependencies = [
+    'isodate>=0.6.0,<0.7.0',
+    'numpy>=1.19.5,<2.0.0',
+    'pycountry>=20.7.3,<21.0.0'
+]
+
+[project.optional-dependencies]
+dev = [
+    'arelle_release==2.4.2',
+    'lxml>=4.6.3,<5.0.0',
+    'mock>=4.0.3,<6.0.0',
+    'nose2>=0.12.0,<1.0.0',
+    'typing-extensions==4.5.0'
+]
```

It appears that the dependencies for ixbrl-viewer haven't been reviewed in a while. The project explicitly imports Arelle, lxml, and pycountry. However, I didn't find any use of numpy or isodate. Given that we have a package for Arelle now, we should list that as an explicit dependency. Also, we no longer need mock, since it's been a part of the standard library since Python 3.3.

What concerns me a bit is that the currently defined pycountry version range in the project is incompatible with the version that's [bundled in our frozen Arelle builds](https://github.com/Arelle/Arelle/blob/5b9f93464d1e7239f7a3abb5d4800182cd1da813/requirements.txt#L13). @paulwarren-wk, are you aware of any compatibility issues between ixbrl-viewer and pycountry version 21 and above?

```suggestion
dependencies = [
    'arelle_release==2.*',
    'lxml==4.*',
    'pycountry==22.*'
]

[project.optional-dependencies]
dev = [
    'nose2==0.13.0',
    'typing-extensions==4.6.3'
]
```

---

A quick look at the [comparison between 20.7.3 and 22.3.5](https://github.com/flyingcircusio/pycountry/compare/20.7.3...22.3.5) wouldn't indicate any breaking changes, it's mostly just formatting and package details with the main functional change being an update to the ISO data.

---

Yeah, seems okay to update